src/main/java/cz/smarteon/loxone/LoxoneWebSocket.java

```diff
@@ -192,20 +195,11 @@ public void setAutoRestart(final boolean autoRestart) {
     }
 
     /**
-     * Set the web socket listener allowing to handle web socket events.
+     * Register the web socket listener allowing to handle web socket events.
      * @param webSocketListener web socket listener, accepts null in order to remove the current listener
      */
-    public void setWebSocketListener(final LoxoneWebSocketListener webSocketListener) {
-        this.webSocketListener = webSocketListener;
-    }
-
-    /**
-     * Get current web socket listener, null by default
-     * @return currently set web socket listener, or null
-     */
-    @Nullable
-    public LoxoneWebSocketListener getWebSocketListener() {
-        return webSocketListener;
+    public void registerWebSocketListener(final @NotNull LoxoneWebSocketListener webSocketListener) {
```

Comment:
Not sure if there exists a use case for that but what about the `unregisterWebSocketListener()` method. That would substitute the `setWebSocketListener(null)`

Reply:
we have no such usecase now, so let's add it when needed